packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -427,15 +432,29 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           }
 
           // normalize
-          const [normalizedUrl, resolvedId] = await normalizeUrl(
+          const [normalizedUrl, resolvedId, resolvedMeta] = await normalizeUrl(
             specifier,
             start
           )
           const url = normalizedUrl
 
           // record as safe modules
-          server?.moduleGraph.safeModulesPath.add(fsPathFromUrl(url))
+          moduleGraph.safeModulesPath.add(fsPathFromUrl(url))
+
+          // ensure module is in the graph under the correct
+          // resolvedId and sporting correct meta properties.
+          const mod = moduleGraph.getModuleById(resolvedId)
+          if (mod) {
+            mod.meta = { ...mod.meta, ...resolvedMeta }
+          } else {
+            moduleGraph.ensureEntryFromResolved([
```

Comment:
This isn't a viable solution, because we want to avoid adding a `ModuleNode` to the module graph *until* it's been successfully loaded. (See https://github.com/vitejs/vite/blob/7f535aca8e8683506b2535c3256292abce7b0615/packages/vite/src/node/server/transformRequest.ts#L208-L210)

Reply:
Well, we could always delete the `ModuleNode` if it fails to load.. :)